examples/CampaignManagement/AddCompleteCampaignsUsingMutateJob.php

```diff
@@ -0,0 +1,483 @@
+        );
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (CampaignOperation $campaignOperation) {
+                return new MutateOperation(['campaign_operation' => $campaignOperation]);
+            },
+            $campaignOperations
+        ));
+
+        // Creates new campaign criterion operations and adds them to the array of mutate
+        // operations.
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (CampaignCriterionOperation $campaignCriterionOperation) {
+                return new MutateOperation(
+                    ['campaign_criterion_operation' => $campaignCriterionOperation]
+                );
+            },
+            self::buildCampaignCriterionOperations($campaignOperations)
+        ));
+
+        // Creates new ad group operations and adds them to the array of mutate operations.
+        $adGroupOperations = self::buildAdGroupOperations($customerId, $campaignOperations);
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (AdGroupOperation $adGroupOperation) {
+                return new MutateOperation(['ad_group_operation' => $adGroupOperation]);
+            },
+            $adGroupOperations
+        ));
+
+        // Creates new ad group criterion operations and adds them to the array of mutate
+        // operations.
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (AdGroupCriterionOperation $adGroupCriterionOperation) {
+                return new MutateOperation(
+                    ['ad_group_criterion_operation' => $adGroupCriterionOperation]
+                );
+            },
+            self::buildAdGroupCriterionOperations($adGroupOperations)
+        ));
+
+        // Creates new ad group ad operations and adds them to the array of mutate operations.
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (AdGroupAdOperation $adGroupAdOperation) {
+                return new MutateOperation(['ad_group_ad_operation' => $adGroupAdOperation]);
+            },
+            self::buildAdGroupAdOperations($adGroupOperations)
+        ));
+
+        return $mutateOperations;
+    }
+
+    /**
+     * Builds a new campaign budget operation for the specified customer ID.
+     *
+     * @param int $customerId the client customer ID
+     * @return CampaignBudgetOperation the campaign budget operation
+     */
+    private static function buildCampaignBudgetOperation(int $customerId)
+    {
+        // Creates a campaign budget operation.
+        return new CampaignBudgetOperation([
+            'create' => new CampaignBudget([
+                // Creates a resource name using the temporary ID.
+                'resource_name' => ResourceNames::forCampaignBudget(
+                    $customerId,
+                    self::$temporaryId--
```

Comment:
Can we extract the temporaryId generation to a small method like getNextTempId()? It's easier to read that way and understand what's happening.

Reply:
Sure.